operations/setup.go

```diff
@@ -130,12 +136,49 @@ func (c *serviceConf) setup(ctx context.Context) error {
 	return nil
 }
 
-func newServiceConf(numWorkers int, localQueue bool, mongodbURI, bucket, dbName string) *serviceConf {
+type dbCreds struct {
+	DBUser string `yaml:"mdb_database_username"`
+	DBPwd  string `yaml:"mdb_database_password"`
+}
+
+func loadCredsFromYAML(filePath string) (*dbCreds, error) {
+	creds := &dbCreds{}
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		grip.Error(err)
+		return nil, err
+	}
+	defer file.Close()
+
+	decoder := yaml.NewDecoder(file)
+
+	if err := decoder.Decode(&creds); err != nil {
+		return nil, err
+	}
+
+	return creds, nil
+}
+
+func newServiceConf(numWorkers int, localQueue bool, mongodbURI, bucket, dbName string, dbCredFile string) *serviceConf {
+
+	creds := &dbCreds{}
+	var err error
+	if dbCredFile != "" {
+
+		creds, err = loadCredsFromYAML(dbCredFile)
```

Comment:
I think this can be `creds, err :=`. I think it's unnecessary to initialize those two variables separately.

Reply:
Creds needs to be initialized outside the if loop or they go out of scope and can't be used in return. If I don't initialize err and use a walrus, it creates a shadowed creds. As written is the only way I was able to get build to work.